khal won't build on master because of failing tests

### Steps To Reproduce
Steps to reproduce the behavior:
1. switch to the `master` branch
2. `nix-build -A khal`

### Build log
```
>       assert create_timezone(bogota, atime, atime).to_ical().split(b'\r\n') == vbogota
>       assert create_timezone(bogota, atime, atime).to_ical().split(b'\r\n') == vbogota
E       AssertionError: assert [b'BEGIN:VTIM...M:-0400', ...] == [b'BEGIN:VTIM...AME:-05', ...]
E       AssertionError: assert [b'BEGIN:VTIM...M:-0400', ...] == [b'BEGIN:VTIM...AME:-05', ...]
E         At index 4 diff: b'TZNAME:-05' != b'RDATE:20380118T221407'
E         At index 4 diff: b'TZNAME:-05' != b'RDATE:20380118T221407'
E         Right contains one more item: b''
E         Right contains one more item: b''
E         Use -v to get more diff
E         Use -v to get more diff


tests/vtimezone_test.py:81: AssertionError
tests/vtimezone_test.py:81: AssertionError
=========================== short test summary info ============================
=========================== short test summary info ============================
FAILED tests/event_test.py::test_event_no_dst - AssertionError: assert ('VCAL...
FAILED tests/event_test.py::test_event_no_dst - AssertionError: assert ('VCAL...
FAILED tests/vtimezone_test.py::test_bogota - AssertionError: assert [b'BEGIN...
FAILED tests/vtimezone_test.py::test_bogota - AssertionError: assert [b'BEGIN...
====== 2 failed, 302 passed, 3 deselected, 1 xfailed, 2 xpassed in 3.05s =======
====== 2 failed, 302 passed, 3 deselected, 1 xfailed, 2 xpassed in 3.05s =======
```

### Notify maintainers

@gebner @fabaff 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.10-200.fc36.x86_64, Fedora Linux, 36 (Workstation Edition), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `""`
 - nixpkgs: `/home/cassou/Documents/projects/nix-system/nixpkgs`
```

It seems like it started to fail again, with a different set of tests:

```sh
error: builder for '/nix/store/bbbcrp1jxfmm1sw2nz6h0n03mjwnhwjm-khal-0.10.5.drv' failed with exit code 1;
       last 10 log lines:
       > FAILED tests/parse_datetime_test.py::test_leap_year - AssertionError: assert b'BEGIN:VEVEN...ND:VEVENT\r\n' == b'BEGIN:VEVEN...ND...
       > FAILED tests/parse_datetime_test.py::test_description - AssertionError: assert b'BEGIN:VEVEN...ND:VEVENT\r\n' == b'BEGIN:VEVEN...ND...
       > FAILED tests/parse_datetime_test.py::test_repeat_floating - AssertionError: assert ('VEVENT', b'..., frozenset()) == ('VEVENT', b'..., ...
       > FAILED tests/parse_datetime_test.py::test_repeat_localized - AssertionError: assert ('VEVENT', b'..., frozenset()) == ('VEVENT', b'..., ...
       > FAILED tests/parse_datetime_test.py::test_alarm - AssertionError: assert b'BEGIN:VEVEN...ND:VEVENT\r\n' == b'BEGIN:VEVEN...ND...
       > FAILED tests/parse_datetime_test.py::test_description_and_location_and_categories - AssertionError: assert b'BEGIN:VEVEN...ND:VEVENT\r\n' == b'BEGIN:VEVEN...ND...
       > FAILED tests/vtimezone_test.py::test_berlin - AssertionError: assert b'BEGIN:STANDARD\r\nDTSTART;VALUE=DATE-TIME:20141026...
       > FAILED tests/vtimezone_test.py::test_berlin_rdate - AssertionError: assert b'BEGIN:STANDARD\r\nDTSTART;VALUE=DATE-TIME:20141026...
       > ===== 19 failed, 282 passed, 6 deselected, 1 xfailed, 2 xpassed in 55.76s ======
       > /nix/store/chilfhdcsnmwjl7igrw26j1lrc0zar35-stdenv-linux/setup: line 1570: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/bbbcrp1jxfmm1sw2nz6h0n03mjwnhwjm-khal-0.10.5.drv'.
```

Maybe we should just disable testing on this package?

--- a/pkgs/applications/misc/khal/default.nix
+++ b/pkgs/applications/misc/khal/default.nix
@@ -1,7 +1,6 @@
 { lib
 , stdenv
 , fetchFromGitHub
-, fetchpatch
 , glibcLocales
 , installShellFiles
 , python3
@@ -9,13 +8,13 @@
 
 python3.pkgs.buildPythonApplication rec {
   pname = "khal";
-  version = "0.10.5";
+  version = "0.11.1";
 
   src = fetchFromGitHub {
     owner = "pimutils";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-FneJmoAOb7WjSgluCwlspf27IU3MsQZFKryL9OSSsUw=";
+    hash = "sha256-5wBKo24EKdEUoYhhv1EqMPOjdwUS31d3R24kLdbyvPA=";
   };
 
   SETUPTOOLS_SCM_PRETEND_VERSION = version;
@@ -53,18 +52,6 @@ python3.pkgs.buildPythonApplication rec {
     packaging
     pytestCheckHook
     vdirsyncer
-  ];
-
-  patches = [
-    # Tests working with latest pytz version, https://github.com/pimutils/khal/pull/1183
-    (fetchpatch {
-      name = "support-later-pytz.patch";
-      url = "https://github.com/pimutils/khal/commit/53eb8a7426d5c09478c78d809c4df4391438e246.patch";
-      sha256 = "sha256-drGtvJlQ3qFUdeukRWCFycPSZGWG/FSRqnbwJzFKITc=";
-      excludes = [
-        "CHANGELOG.rst"
-      ];
-    })
   ];
 
   postInstall = ''

